modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,375 @@
+ */
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @param array $metadata The attachment metadata.
+ * @param int   $attachment_id The attachment ID.
+ *
+ * @return array $metadata
+ * @since n.e.x.t
+ */
+function dominant_color_metadata( $metadata, $attachment_id ) {
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+
+		return $metadata;
+	}
+
+	$dominant_color = dominant_color_get( $attachment_id );
+
+	if ( ! empty( $dominant_color ) ) {
+		$metadata['dominant_color'] = $dominant_color;
+	}
+
+	return $metadata;
+}
+
+add_filter( 'wp_generate_attachment_metadata', 'dominant_color_metadata', 10, 2 );
+
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @param array $metadata Metadata for the attachment.
+ * @param int   $attachment_id attachement id.
+ *
+ * @return array $metadata
+ * @since n.e.x.t
+ */
+function dominant_color_has_transparency_metadata( $metadata, $attachment_id ) {
+
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+		return $metadata;
+	}
+
+	$has_transparency = dominant_color_get_has_transparency( $attachment_id );
+
+	if ( ! empty( $has_transparency ) ) {
+		$metadata['has_transparency'] = $has_transparency;
+	} else {
+		$metadata['has_transparency'] = false;
+	}
+
+	return $metadata;
+}
+
+add_filter( 'wp_generate_attachment_metadata', 'dominant_color_has_transparency_metadata', 10, 2 );
+
+/**
+ * Filter various image attributes to add the dominant color to the image
+ *
+ * @param array  $attr Attributes for the image markup.
+ * @param object $attachment Image attachment post.
+ *
+ * @return mixed
+ * @since n.e.x.t
+ */
+function dominant_color_tag_add_adjust_to_image_attributes( $attr, $attachment ) {
+
+	$image_meta = wp_get_attachment_metadata( $attachment->ID );
+	if ( ! is_array( $image_meta ) ) {
+		return $attr;
+	}
+
+	$has_transparency = isset( $image_meta['has_transparency'] ) ? $image_meta['has_transparency'] : false;
+
+	$extra_class = '';
+	if ( ! isset( $attr['style'] ) ) {
+		$attr['style'] = ' --has-transparency: ' . $has_transparency . '; ';
+	} else {
+		$attr['style'] .= ' --has-transparency: ' . $has_transparency . '; ';
+		$extra_class    = ' has-transparency ';
+	}
+
+	if ( isset( $image_meta['dominant_color'] ) ) {
+
+		$attr['data-dominant-color'] = $image_meta['dominant_color'];
+
+		if ( empty( $attr['style'] ) ) {
+			$attr['style'] = '';
+		}
+		$attr['style'] .= '--dominant-color: #' . $image_meta['dominant_color'] . ';';
+
+		$extra_class .= ( dominant_color_color_is_light( $image_meta['dominant_color'] ) ) ? 'dominant-color-light' : 'dominant-color-dark';
+		if ( empty( $attr['class'] ) ) {
+			$attr['class'] = '';
+		}
+		if ( isset( $attr['class'] ) && ! array_intersect( explode( ' ', $attr['class'] ), explode( ' ', $extra_class ) ) ) {
+			$attr['class'] .= ' ' . $attr['class'];
+		}
+	}
+
+	return $attr;
+}
+
+add_filter( 'wp_get_attachment_image_attributes', 'dominant_color_tag_add_adjust_to_image_attributes', 10, 2 );
```

Comment:
See above. Please update this throughout the PR.
```suggestion
}
add_filter( 'wp_get_attachment_image_attributes', 'dominant_color_tag_add_adjust_to_image_attributes', 10, 2 );
```